src/doc/unstable-book/src/language-features/macro-metavar-expr-concat.md

```diff
@@ -0,0 +1,154 @@
+# `macro_metavar_expr_concat`
+
+The tracking issue for this feature is: [#124225]
+
+------------------------
+
+
+`#![feature(macro_metavar_expr_concat)]` provides a more powerful alternative to [`concat_idents!`].
+
+> This feature is not to be confused with [`macro_metavar_expr`] or [`concat_ident`].
+
+> This is an experimental feature; it and its syntax will require a RFC before stabilization.
+
+
+### Overview
+
+`macro_rules!` macros cannot create new identifiers and use them in ident positions.
+A common use case is the need to create new structs or functions. The following cannot be done[^1]:
+
+```rust,compile_fail
+macro_rules! create_some_structs {
+  ($name:ident) => {
+      // Invalid syntax
+      struct First_$name;
+       // Also invalid syntax
+      struct Second_($name);
+      // Macros are not allowed in this position
+      // (This restriction is what makes `concat_idents!` useless)
+      struct concat_ident!(Third_, $name);
+  }
+}
+# create_some_structs!(Thing);
+```
+
+`#![feature(macro_metavar_expr_concat)]` provides the `concat` metavariable to concatenate idents in ident position:
```

Comment:
same comment regarding "in ident position".

Reply:
"in ident positions" can probably just be dropped. I think it is meant to address the comparison to `concat_idents!` which can't be used in these positions, but that comparison isn't as meaningful with `concat_idents!` going away.